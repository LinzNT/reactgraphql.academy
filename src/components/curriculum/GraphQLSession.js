import React from 'react'
import Ul, { Li } from '../layout/Ul'
import Session from './Session'

const GraphQLSession = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>
        GraphQL
        <Ul>
          <Li>
            Understand the fundamental differences between a REST API and a
            GraphQL API
          </Li>
          <Li>GraphQL Queries & Mutations</Li>
          <Li>Relay Connections</Li>
        </Ul>
      </Li>
      <Li>
        Apollo client
        <Ul>
          <Li>Connecting an Apollo client to a GraphQL server</Li>
          <Li>
            Implement infinite scrolling in a React app using Apollo on the
            client-side and a Relay connection on the server-side
          </Li>
          <Li>
            Update the state of your React app using mutations and Apollo client
          </Li>
        </Ul>
      </Li>
    </Ul>
  </Session>
)

export default GraphQLSession
