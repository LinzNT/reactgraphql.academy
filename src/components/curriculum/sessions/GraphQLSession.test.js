import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import GraphQLSession from './GraphQLSession'

describe('<GraphQLSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<GraphQLSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
