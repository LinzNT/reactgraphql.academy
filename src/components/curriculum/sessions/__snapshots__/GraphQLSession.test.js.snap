// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraphQLSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      GraphQL
      <styled.ul>
        <styled.li>
          Understand the fundamental differences between a REST API and a GraphQL API
        </styled.li>
        <styled.li>
          GraphQL Queries & Mutations
        </styled.li>
        <styled.li>
          Relay Connections
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Apollo client
      <styled.ul>
        <styled.li>
          Connecting an Apollo client to a GraphQL server
        </styled.li>
        <styled.li>
          Implement infinite scrolling in a React app using Apollo on the client-side and a Relay connection on the server-side
        </styled.li>
        <styled.li>
          Update the state of your React app using mutations and Apollo client
        </styled.li>
      </styled.ul>
    </styled.li>
  </styled.ul>
</Session>
`;
